Sources/NetworkProtection/KeyManagement/NetworkProtectionKeyStore.swift

```diff
@@ -37,15 +37,6 @@ public protocol NetworkProtectionKeyStore {
     ///
     func updateKeyPair(_ newKeyPair: KeyPair)
 
-    /// Updates the current `KeyPair` to have the specified expiration date
-    ///
-    /// - Parameters:
-    ///     - newExpirationDate: the new expiration date for the KeyPair
-    ///
-    /// - Returns: a new KeyPair with the specified updates
-    ///
-    func updateKeyPairExpirationDate(_ newDate: Date) -> KeyPair
```

Comment:
This isn't great anymore, because the current keypair can be `nil`, so this method may not have anything to return.

In the actual case where this was being called, the key pair should really not be `nil`, so there's a conflict between caller and implementation.

To make things a bit cleaner, since the caller already has a valid keypair, I'm opting to just allow the caller to update the keypair data as desired (through `updateKeyPair`).  This helps us go in the direction of the store being "dumber" but also helps us maintain cleaner code.